lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystem.java

```diff
@@ -58,6 +58,8 @@ final class S3FileSystem
     private final S3Context context;
     private final RequestPayer requestPayer;
 
+    private static final S3FileSystemStats STATS = new S3FileSystemStats();
```

Comment:
This should be injected in the constructor. The reason it is static in `TrinoS3FileSystem` is because Hadoop file systems are not constructed by Guice.

Reply:
Thank you. Updated in the revision.